frontend/packages/dev-console/src/components/topology2/components/nodes/EventSource.tsx

```diff
@@ -0,0 +1,83 @@
+import * as React from 'react';
+import {
+  Node,
+  observer,
+  useHover,
+  WithSelectionProps,
+  WithContextMenuProps,
+  useSvgAnchor,
+  useCombineRefs,
+  WithDragNodeProps,
+  createSvgIdUrl,
+} from '@console/topology';
+import SvgBoxedText from '../../../svg/SvgBoxedText';
+import NodeShadows, { NODE_SHADOW_FILTER_ID_HOVER, NODE_SHADOW_FILTER_ID } from '../NodeShadows';
+
+import './EventSource.scss';
+
+export type EventSourceProps = {
+  element: Node;
+} & WithSelectionProps &
+  WithDragNodeProps &
+  WithContextMenuProps;
+
+const EventSource: React.FC<EventSourceProps> = ({
+  element,
+  selected,
+  onSelect,
+  onContextMenu,
+  dragNodeRef,
+}) => {
+  const svgAnchorRef = useSvgAnchor();
+  const [hover, hoverRef] = useHover();
+  const groupRefs = useCombineRefs(dragNodeRef, hoverRef);
+  const { width, height } = element.getBounds();
+  const size = Math.min(width, height);
+  const { data } = element.getData();
+
+  return (
+    <g onClick={onSelect} onContextMenu={onContextMenu} ref={groupRefs}>
+      <NodeShadows />
+      <polygon
+        className="odc-event-source"
+        ref={svgAnchorRef}
+        filter={createSvgIdUrl(hover ? NODE_SHADOW_FILTER_ID_HOVER : NODE_SHADOW_FILTER_ID)}
+        points={`${width / 2}, ${(height - size) / 2} ${width - (width - size) / 2},${height /
+          2} ${width / 2},${height - (height - size) / 2} ${(width - size) / 2},${height / 2}`}
+      />
+
+      {selected && (
+        <polygon
+          className="odc-event-source__selection"
+          points={`${width / 2}, ${(height - size) / 2 - 2} ${width -
+            (width - size) / 2 +
+            2},${height / 2} ${width / 2},${height - (height - size) / 2 + 2} ${(width - size) / 2 -
+            2},${height / 2}`}
+        />
+      )}
+      <image
+        x={width * 0.25}
+        y={height * 0.25}
+        width={size * 0.5}
+        height={size * 0.5}
+        // TODO replace with icon based on data
+        xlinkHref="static/assets/openshift.svg"
```

Comment:
Was this done by the other eventing work and lost during translation?

Reply:
@invincibleJai has been working on event sources on his branch.

Reply:
Will file an issue to have image icons corresponding to various event sources , once this goes in